others/sub_array_with_given_sum.cpp

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2020 @author omkarlanghe
+ *
+ * @file
+ * Given an unsorted array A of size N of non-negative integers,
+ * find a continuous sub-array which adds to a given number S, If not found
+print -1.
+ *
+ * Input:
+    3
+    5 12
+    1 2 3 7 5
+    5 123
+    1 2 3 7 5
+    10 15
+    1 2 3 4 5 6 7 8 9 10
+* Output:
+    2 4
+    -1
+    1 5
+* Explanation :
+    Testcase1: sum of elements from 2nd position to 4th position is 12.
+(consider 1 based indexing) Testcase2: no subarray found whose sum equals to the
+given sum, hence print -1. Testcase3: sum of elements from 1st position to 5th
+position is 15. (consider 1 based indexing)
+*/
+
+#include <iostream>
+/**
+ * Function to print the index range of subarray equal to the given sum. If no
+ * such subarray found, print -1.
+ * @param arr array to check
+ * @param size size of an array
+ * @param original_sum sum to match with the sum of all elements of a sub array.
+ * @return void and prints index ranges if sum of sub array equal to the given
+ * sum, else prints -1.
+ */
+void contiguous_sub_array(int *arr, int size, int orignal_sum) {
```

Comment:
The sum should be returned as function return value. Printing it should be done by the calling function.